includes/payment-methods/class-upe-payment-gateway.php

```diff
@@ -577,7 +604,14 @@ public function process_payment( $order_id ) {
 				}
 			}
 		} else {
-			return $this->parent_process_payment( $order_id );
+			try {
+				return $this->parent_process_payment( $order_id );
+			} catch ( Exception $e ) {
+				// The invoked method already handles order status changes on success and partially on failure.
+				// But, not all exceptions are triggering order status changes, so we have to clear intent cache.
+				// The intent caching is UPE-specific, so we have to keep it in the UPE-gateways only.
+				self::remove_upe_payment_intent_from_session();
```

Comment:
```suggestion
				self::remove_upe_payment_intent_from_session();
				throw $e;
```
Do we not need to rethrow the exception here as well? Otherwise if the parent gateway class throws an exception, for whatever reason, won't we just catch it and then continue on to return sucess instead of failure?